presto-main/src/main/java/com/facebook/presto/operator/aggregation/arrayagg/ArrayAggregationStateFactory.java

```diff
@@ -26,7 +28,9 @@ public class ArrayAggregationStateFactory
     private final Type type;
     private final ArrayAggGroupImplementation mode;
 
-    public ArrayAggregationStateFactory(Type type, ArrayAggGroupImplementation mode)
+    public ArrayAggregationStateFactory(
+            @TypeParameter("T") Type type,
+            @AggregationState ArrayAggGroupImplementation mode)
```

Comment:
This constructor parameter isn't supported on implementations of the `AccumulatorStateFactory` interface. We'll either need to drop the support for the aggregation mode config, or find another way to access arbitrary configs like this one.